src/test/java/com/waffle/dredes/gameobject/player/Player.java

```diff
@@ -57,35 +60,65 @@ public void start() {
             transform = new TransformComponent(400, 0);
             kinematics = new KinematicComponent(new Vec2f(), new Vec2f(), 0, 1);
-            gc = new ColliderComponent(new Vec2f(0, b.getHeight()), new Vec2f(b.getWidth(), 1) , e -> {
+            gc = new ColliderComponent(new Vec2f(0, 0), new Vec2f(b.getWidth(), b.getHeight()) , e -> {
                 if(e.getCollidedObject() instanceof Tile tile) {
-                    Vec2f mid = new Vec2f(transform.position.x + b.getWidth() / 2f, transform.position.y + b.getHeight() / 2f);
-                    if(kinematics.v.y > 0) {
-                        // Ceiling
-                        kinematics.v.y = 0;
-                        transform.position.y = tile.transform.position.y + tile.height;
-                    } else if(kinematics.v.y < 0) {
-                        // Floor
-                        kinematics.v.y = 0;
-                        transform.position.y = tile.transform.position.y;
-                    }
-                    if(kinematics.v.x > 0) {
-                        // Wall on the right of player
-                        kinematics.v.x = 0;
-                        transform.position.x = tile.transform.position.x;
-                    } else if(kinematics.v.x < 0) {
-                        // Wall on the left of player
-                        kinematics.v.x = 0;
-                        transform.position.x = tile.transform.position.x + tile.width;
+                    groundCheck = true;
+                    if(!tile.fluid)
+                    {
+
+                        float slope = ((transform.position.y +((float)b.getHeight()/2))-(tile.transform.position.y + 16))/((transform.position.x - ((float)b.getWidth()/2))-(tile.transform.position.x + 16));
+                        if(Math.abs(slope) < (float)b.getHeight()/b.getWidth())
+                        {
+                            if(transform.position.x > tile.transform.position.x)
+                            {
+                                wallCheckL = true;
+                            }
+                            else{
+                                wallCheckR = true;
+                            }
+                        }
+                        else if(transform.position.y > tile.transform.position.y)
+                        {
+                            ceilingCheck = true;
+                            System.out.println("Ceiling");
+                            if(current instanceof Jumping && current.active)
+                            {
+                                current.active = false;
+                            }
+                        }
+                        else
+                        {
+                            groundCheck = true;
+                        }
+
                     }
+//                    Vec2f mid = new Vec2f(transform.position.x + b.getWidth() / 2f, transform.position.y + b.getHeight() / 2f);
```

Comment:
You should really remove the code instead of commenting it out. You can use source control to restore it if need be.

Reply:
Both of us are relatively new to the multiple-contributor workflow, we'll definitely add this on to our list of features to know! :)

Reply:
Not overly pertinent in your own code base but something to go by: [Clean Code](https://thixalongmy.haugiang.gov.vn/media/1175/clean_code.pdf)